pluggy.py

```diff
@@ -640,7 +643,10 @@ def varnames(func, startindex=None):
         startindex = 1
     else:
         if not inspect.isfunction(func) and not inspect.ismethod(func):
-            func = getattr(func, '__call__', func)
+            try:
```

Comment:
this is missing a big part of the rest of the matching, we should either have varnames fail badly and ensure we dont pass it bad things or make it truely resilent

we should also disengage the varname caching and the varname obtaining, 
we very likely do not really need the cache and its an extra source of errors